win32/devdoc/srw_lock_ll_win32_requirements.md

```diff
@@ -0,0 +1,127 @@
+# `srw_lock_ll_win32` requirements
+
+## Overview
+
+`srw_lock_ll_win32` is the implementation of a slim reader writer lock (`srw_lock_ll` interface). On Windows it simply wraps `SRWLOCK`.
+
+## Exposed API
+
+```c
+#define SRW_LOCK_LL_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_LL_TRY_ACQUIRE_OK, \
+    SRW_LOCK_LL_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_LL_TRY_ACQUIRE_INVALID_ARGS
+
+typedef void* SRW_LOCK_LL;
```

Comment:
>void [](http://example.com/codeflow?start=8&length=4)

when we are in Windows land here, `SRWLOCK` should appear someplace in this file... as an explicit type.

Reply:
Myeah, I went back and forth with this with myself. This is why I needed the review actually and I am glad it got picked on.

I made SRL_LOCK_LL now 

#ifdef WIN32
typedef SRWLOCK SRW_LOCK_LL;
#endif